Content.Server/Radio/EntitySystems/JammerSystem.cs

```diff
@@ -30,39 +33,65 @@ public override void Initialize()
         SubscribeLocalEvent<ActiveRadioJammerComponent, PowerCellChangedEvent>(OnPowerCellChanged);
         SubscribeLocalEvent<RadioJammerComponent, ExaminedEvent>(OnExamine);
         SubscribeLocalEvent<RadioSendAttemptEvent>(OnRadioSendAttempt);
+        SubscribeLocalEvent<RadioJammerComponent, GetVerbsEvent<Verb>>(OnGetVerb);
     }
 
     public override void Update(float frameTime)
     {
         var query = EntityQueryEnumerator<ActiveRadioJammerComponent, RadioJammerComponent>();
+
         while (query.MoveNext(out var uid, out var _, out var jam))
         {
-            if (_powerCell.TryGetBatteryFromSlot(uid, out var batteryUid, out var battery) &&
-                !_battery.TryUseCharge(batteryUid.Value, jam.Wattage * frameTime, battery))
+
+            if (_powerCell.TryGetBatteryFromSlot(uid, out var batteryUid, out var battery))
             {
-                RemComp<ActiveRadioJammerComponent>(uid);
-                RemComp<DeviceNetworkJammerComponent>(uid);
+                if (!_battery.TryUseCharge(batteryUid.Value, GetCurrentWattage(jam) * frameTime, battery))
+                {
+                    ChangeLEDState(false, uid);
```

Comment:
This looks like it can just use powercelldraw.

Reply:
It could but after looking at powercelldraw I don't think it is the correct approach!

1.) The texture needs to update depending on the charge so Update will still have to be in the jammer system.

2.) Because each power level has a unique wattage, I would have to dynamically update the powercelldraw component either every update tick (Then its literally like the same as it is now just weirder) or in the verb act event (It will be ugly and kind of confusing).

I'll obviously still do it if you want me to but unless you can think of a way to resolve these issues I don't think think its a good idea.

Let me know what you think!